src/libstd/collections/hash/table.rs

```diff
@@ -34,6 +34,37 @@ type HashUint = usize;
 
 const EMPTY_BUCKET: HashUint = 0;
 
+/// Special `Unique<T>` that uses the lower bit of the pointer to expose
+/// a boolean tag. Can only be used with T's with alignment > 1.
+struct TaggedUnique<T>(Unique<T>);
+
+impl<T> TaggedUnique<T> {
+    unsafe fn new(ptr: *mut T) -> Self {
+        assert!(mem::align_of::<T>() > 1);
```

Comment:
So, a few comments here. 

While I don’t think that happens in *practice*, it is certainly possible for `HashUint` get an alignment of `1` in *theory*. For example a custom data-layout could be used to achieve that. This is *one* minor portability concern.

I’m somewhat doubting the need to have this as a generic pointer, rather than just a `TaggedHashUintPtr` type. This is so unusual a thing to do, that having a generic version is unlikely to ever have any benefit.